File: spa/src/view-models/character.view-model.ts
import { ItemViewModel } from "./item.view-model";
import { CreatureSizes, CreatureTraitViewModel, ResistanceViewModel, SpellcastingViewModel } from "./shared.view-model";

export interface ClassViewModel {
    name: string;
    level: number;
}

export class CharacterViewModel {
    id: string = '';
    name: string = '';
    size: CreatureSizes = CreatureSizes.Unknown;
    level: number = 0;
    race: string = '';
    classes: ClassViewModel[] = [];
    hitpointMaximum: number = 0;
    proficiencyBonus: number = 0;
    walkingSpeed: number = 0;
    climbingSpeed: number = 0;
    burrowingSpeed: number = 0;
    swimmingSpeed: number = 0;
    flyingSpeed: number = 0;
    canHover: boolean = false;
    speedConditions: string[] = [];
    attributeStr: number = 0;
    attributeDex: number = 0;
    attributeCon: number = 0;
    attributeInt: number = 0;
    attributeWis: number = 0;
    attributeCha: number = 0;
    skillProficiencies: string[] = [];
    savingThrowProficiencies: string[] = [];
    passivePerception: number = 0;
    resistances: ResistanceViewModel[] = [];
    immunities: ResistanceViewModel[] = [];
    languages: string[] = [];
    traits: CreatureTraitViewModel[] = [];
    actions: CreatureTraitViewModel[] = [];
    reactions: CreatureTraitViewModel[] = [];
    spellcasting: SpellcastingViewModel[] = [];
    equipment: ItemViewModel[] = [];
    inventory: ItemViewModel[] = [];
    senses: string[] = [];
}

File: backends/DDD.orch3strator/ViewModels/DnD5e/CharacterViewModel.cs
using System.Text.Json.Serialization;

namespace DDD.orch3strator.ViewModels.DnD5e
{
  public class CharacterViewModel : CharacterBaseViewModel
  {
    [JsonPropertyName("armourClass")]
    public ArmourClassViewModel ArmourClass { get; set; }

    [JsonPropertyName("classes")]
    public CharacterClassViewModel Classes { get; set; }

    [JsonPropertyName("hitpointMaximum")]
    public int HitpointMaximum { get; set; }

    [JsonPropertyName("walkingSpeed")]
    public int WalkingSpeed { get; set; }

    [JsonPropertyName("flyingSpeed")]
    public int FlyingSpeed { get; set; }

    [JsonPropertyName("swimmingSpeed")]
    public int SwimmingSpeed { get; set; }

    [JsonPropertyName("climbingSpeed")]
    public int ClimbingSpeed { get; set; }

    [JsonPropertyName("burrowingSpeed")]
    public int BurrowingSpeed { get; set; }

    [JsonPropertyName("attributeStr")]
    public int AttributeStr { get; set; }

    [JsonPropertyName("attributeDex")]
    public int AttributeDex { get; set; }

    [JsonPropertyName("attributeCon")]
    public int AttributeCon { get; set; }

    [JsonPropertyName("attributeInt")]
    public int AttributeInt { get; set; }

    [JsonPropertyName("attributeWis")]
    public int AttributeWis { get; set; }

    [JsonPropertyName("attributeCha")]
    public int AttributeCha { get; set; }
  }

  public class CharacterClassViewModel
  {
    [JsonPropertyName("name")]
    public string Name { get; set; }

    [JsonPropertyName("level")]
    public int Level { get; set; }
  }
}
